test/avio.jl

```diff
@@ -214,13 +240,69 @@ end
             encodedvideopath = VideoIO.encodevideo("testvideo.mp4",imgstack,framerate=30,AVCodecContextProperties=props, silent=true)
             @test stat(encodedvideopath).size > 100
             f = VideoIO.openvideo(encodedvideopath)
-            @test VideoIO.counttotalframes(f) == n-4 # videos encoded with crf > 0 have 4 fewer frames
+            @test_broken VideoIO.counttotalframes(f) == n # missing frames due to edit list bug?
             close(f)
             rm(encodedvideopath)
         end
     end
 end
 
+@testset "Simultaneous encoding and muxing" begin
+    testvid = "testvideo.mp4"
+    n = 100
+    encoder_settings = (color_range = 2,)
+    container_private_settings = (movflags = "+write_colr",)
+    for el in [Gray{N0f8}, Gray{N6f10}, RGB{N0f8}, RGB{N6f10}]
+        for scanline_arg in [true, false]
+            @testset "Encoding $el imagestack, scanline_major = $scanline_arg" begin
+                img_stack = map(x -> rand(el, 100, 100), 1 : n)
+                lossless = el <: Gray
+                crf = lossless ? 0 : 20
```

Comment:
Very very minor point, but I believe the default is 23, so we should probably use that here and elsewhere
```suggestion
                crf = lossless ? 0 : 23
```